pydantic/_internal/_typing_extra.py

```diff
@@ -280,13 +280,7 @@ def _make_forward_ref(
 
         Implemented as EAFP with memory.
         """
-        global _make_forward_ref
-        try:
-            res = typing.ForwardRef(arg, is_argument, is_class=is_class)  # type: ignore
-            _make_forward_ref = typing.ForwardRef  # type: ignore
-            return res
-        except TypeError:
-            return typing.ForwardRef(arg, is_argument)
+        return typing.ForwardRef(arg, is_argument)
 
 else:
     _make_forward_ref = typing.ForwardRef
```

Comment:
Do we even need this if/else clause then?

Reply:
I think we need because we are using `_make_forward_ref` with `is_class` in some places and `is_class` is not available in `python < 3.9`

Did I miss something?

Reply:
Gotcha yes makes sense 👍🏻